mongo/change_stream.go

```diff
@@ -107,6 +107,11 @@ func newChangeStream(ctx context.Context, config changeStreamConfig, pipeline in
 		ctx = context.Background()
 	}
 
+	cursorOpts := config.client.createBaseCursorOptions()
+
+	cursorOpts.Registry = config.registry
+	cursorOpts.MarshalValueEncoderFn = newEncoderFn(config.bsonOpts)
```

Comment:
Optional: Consider embedding the `Registry` into the `EncoderFn` instead of passing it as a separate field. Then you can remove the `Registry` field from `CursorOptions`. See related comments on `newEncoderFn` and `codecutil.MarshalValue`.

E.g.
```go
cursorOpts.MarshalValueEncoderFn = newEncoderFn(config.bsonOpts, config.registry)
```